src/app/constructor/constructor.routing.ts

```diff
@@ -2,15 +2,25 @@ import { RouterModule } from '@angular/router';
 import { ConstructorComponent } from './constructor.component';
 import { DeviceListResolver } from '../shared/devices/devices.resolver';
 import { CanDeactivateConstructor } from './can-deactivate-constructor';
+import { ViewResolveService } from './view-resolve.service';
 
 export const routes = [
     {
         path: '',
         component: ConstructorComponent,
-        canDeactivate: [ CanDeactivateConstructor ],
+        canDeactivate: [CanDeactivateConstructor],
         resolve: {
             sensors: DeviceListResolver
         }
+    },
+    {
+        path: 'editor/:id',
```

Comment:
Can we use just ':id' path?
"/constructor" - for creating
"/constructor/:id" - for editing